qadence/model.py

```diff
@@ -432,6 +432,56 @@ def _from_dict(cls, d: dict, as_torch: bool = False) -> QuantumModel:
 
         return qm
 
+    def load_params_from_dict(self, d: dict, strict: bool = True) -> None:
+        """Copy parameters from dictionary into this QuantumModel.
+
+        Unlike :meth:`~qadence.QuantumModel.from_dict`, this method does not create a new
+        QuantumModel instance, but rather loads the parameters into the same QuantumModel.
+        The behaviour of this method is similar to :meth:`~torch.nn.Module.load_state_dict`.
+
+        The dictionary is assumed to have the format as saved via
+        :meth:`~qadence.QuantumModel.to_dict`
+
+        Args:
+            d (dict): The dictionary
+            strict (bool, optional):
+                Whether to strictly enforce that the parameter keys in the dictionary and
+                in the model match exactly. Default: ``True``.
+        """
+        param_dict = d["param_dict"]
+        missing_keys = []
+        unexpected_keys = []
+        for n, param in param_dict.items():
+            if n in self._params.keys():
+                try:
+                    with torch.no_grad():
+                        self._params[n].copy_(
+                            torch.nn.Parameter(param, requires_grad=param.requires_grad)
+                        )
+                except Exception as e:
+                    logger.warning(f"Unable to load parameter {n} from dictionary due to {e}.")
+            else:
+                unexpected_keys.append(n)
+
+        for n in self._params.keys():
```

Comment:
You can use here `missing_keys = set(self._params.keys()) - set(param_dict.keys())`